lib/src/main/java/org/jvnet/localizer/ResourceProvider.java

```diff
@@ -0,0 +1,47 @@
+package org.jvnet.localizer;
+
+import java.net.URL;
+
+/**
+ * Allows customizing what class loader is used to find a given resource.
+ *
+ * By default, it's the class loader for the class of the generated resource bundle, but certain
+ * localization patterns might require a different source.
+ */
+public abstract class ResourceProvider {
+    /**
+     * Return the resource URL specified by the given String
+     */
+    public abstract URL getResource(String name, Class<?> resourceBundle);
+
+    public static void setProvider(ResourceProvider p) {
```

Comment:
Add API annotations instead?

Reply:
Is this the upstream comment for https://github.com/jenkinsci/localization-zh-cn-plugin/pull/31#discussion_r254232469, or what does this mean?

Reply:
This library does not have nullability annotations.